realm/realm-library/src/main/java/io/realm/internal/fields/CachedFieldDescriptor.java

```diff
@@ -82,8 +82,8 @@ protected void compileFieldDescription(List<String> fields) {
             columnType = tableInfo.getColumnType(columnName);
             if (i < nFields - 1) {
                 verifyInternalColumnType(currentTable, columnName, columnType);
-                currentTable = tableInfo.getLinkedTable(columnName);
             }
+            currentTable = tableInfo.getLinkedTable(columnName);
```

Comment:
Does this work?  I absolutely don't remember but that was in the conditional for a reason, at one time.  What happens if you call `schema.getNativeTablePtr(null)` a couple lines down?